parley/src/layout/line/greedy.rs

```diff
@@ -759,8 +760,11 @@ fn try_commit_line<B: Brush>(
                     cluster_range.end = state.clusters.end;
                 }
 
+                let overlap = run_data.cluster_range.contains(&cluster_range.start)
+                    && run_data.cluster_range.contains(&(cluster_range.end - 1));
                 if cluster_range.start > cluster_range.end
                     || (!is_empty && cluster_range.start == cluster_range.end)
+                    || !overlap
```

Comment:
Could this be simplified to the following?

```suggestion
                if cluster_range.start >= run_data.cluster_range.end
                    || (!is_empty && cluster_range.start == cluster_range.end)
```

The idea is that if our next cluster index (`state.clusters.start`) is outside of this run, we've already fully consumed the run. I think where it goes wrong with this bug is that `cluster_range.end` is also updated in the case where there's a single inline box on this line.